File: go/cars-assemble/cars_assemble.go
package cars

import "math"

const IndividualCost = 10000
const BatchCost = 95000
const BatchSize = 10

// CalculateWorkingCarsPerHour calculates how many working cars are
// produced by the assembly line every hour.
func CalculateWorkingCarsPerHour(productionRate int, successRate float64) float64 {
	return float64(productionRate) * successRate / 100
}

// CalculateWorkingCarsPerMinute calculates how many working cars are
// produced by the assembly line every minute.
func CalculateWorkingCarsPerMinute(productionRate int, successRate float64) int {
	return int(math.Floor((CalculateWorkingCarsPerHour(productionRate, successRate) / float64(60.0))))
}

// CalculateCost works out the cost of producing the given number of cars.
func CalculateCost(carsCount int) uint {
	individualProduction := carsCount % BatchSize
	batchCost := (carsCount - individualProduction) / BatchSize * BatchCost
	return uint(batchCost + individualProduction*IndividualCost)
}

File: clojure/cars-assemble/src/cars_assemble.clj
(ns cars-assemble)

(def cars-per-hour 221.0)

(defn success-rate
  "Returns the assembly line's success rate,
   taking into account its speed"
  [speed]
 (cond
   (= 0 speed)      0.00
   (<= speed 4)     1.00
   (<= speed 8)     0.90
   (= 9 speed)      0.80
   (= 10 speed)     0.77
   :else (throw (Exception. "Value should be between 0 and 10"))
   )
  )

(defn production-rate
  "Returns the assembly line's production rate per hour,
   taking into account its success rate"
  [speed]
  (* speed cars-per-hour (success-rate speed)))

(defn working-items
  "Calculates how many working cars are produced per minute"
  [speed]
  (int (/ (production-rate speed) 60)))
